go-controller/pkg/node/gateway.go

```diff
@@ -268,6 +268,35 @@ func gatewayInitInternal(nodeName, gwIntf, egressGatewayIntf string, gwNextHops
 		return nil, nil, err
 	}
 
+	// Set annotation that determines if options:gateway_mtu shall be set for this node.
+	var disableGatewayMtu bool
+	chkPktLengthSupported, err := util.DetectCheckPktLengthSupport(gatewayBridge.bridgeName)
+	if err != nil {
+		return nil, nil, err
+	}
+	if !chkPktLengthSupported {
+		klog.Warningf("Disabling LRP gateway_mtu as this node does not support check_packet_length action")
+		disableGatewayMtu = true
+	}
+	/* This is a work around. In order to have the most optimal performance, the packet MTU check should be
+	 * disabled when OVS HW Offload is enabled on the node. The reason is that OVS HW Offload does not support
+	 * packet MTU checks properly without the offload support for sFlow.
+	 * The patches for sFlow in OvS: https://patchwork.ozlabs.org/project/openvswitch/list/?series=290804
+	 * As of writing these offload support patches for sFlow are in review.
+	 * TODO: This workaround should be removed once the offload support for sFlow patches are merged upstream OvS.
+	 */
+	ovsHardwareOffloadEnabled, err := util.IsOvsHwOffloadEnabled()
+	if err != nil {
+		return nil, nil, err
+	}
+	if ovsHardwareOffloadEnabled {
+		klog.Warningf("Disabling LRP gateway_mtu as hardware offloading is enabled on this node")
+		disableGatewayMtu = true
```

Comment:
this should all be in an else, since there is no point in exeucting it if the previous if statement was true

Reply:
done